rules/Arguments/Rector/ClassMethod/ArgumentAdderRector.php

```diff
@@ -249,8 +257,18 @@ private function shouldSkipParameter(
             return $this->changedArgumentsDetector->isTypeChanged($param, $argumentAdder->getArgumentType());
         }
 
-        if (isset($node->args[$position])) {
-            return true;
+        // If named arguments exist
```

Comment:
I think to get correct position, you can combine it with `argumentName` from:

```php
$position = $this->argsAnalyzer->resolveArgPosition($args, $argumentName, $position);
```

see https://github.com/rectorphp/rector-src/blob/2e1e9cc133451461eff580fd360dffa75d94c03d/src/NodeAnalyzer/ArgsAnalyzer.php#L35-L50

Reply:
Ok, but not sure how that would be useful here

Reply:
That can be useful in the rule that tries to replace an argument, but I don't think we need this in this rule which just tries to add a new argument, if the argument already exists we need to skip the rule and if it doesn't, trying to add it positionally can be problematic, so better just add it as a new named argument